src/layers/ParksLayer/ParksLayer.js

```diff
@@ -0,0 +1,111 @@
+import TileLayer from 'ol/layer/Tile';
+import VectorLayer from 'ol/layer/Vector';
+import Feature from 'ol/Feature';
+import MultiPolygon from 'ol/geom/MultiPolygon';
+import { transform } from 'ol/proj';
+import { Fill, Stroke, Style } from 'ol/style';
+import TrafimageGeoServerWMSLayer from '../TrafimageGeoServerWMSLayer';
+
+class ParksLayer extends TrafimageGeoServerWMSLayer {
+  constructor(options = {}) {
+    super({
+      ...options,
+    });
+    this.abortController = new AbortController();
+
+    this.dataProjection = 'EPSG:21781';
+    this.featureProjection = 'EPSG:3857';
+
+    // Tile layer to display the features.
+    this.tileLayer = this.olLayer
+      .getLayers()
+      .getArray()
+      .find((l) => l instanceof TileLayer);
+    // Vector layer to highlight the hovered feature.
+    this.vectorLayer = this.olLayer
+      .getLayers()
+      .getArray()
+      .find((l) => l instanceof VectorLayer);
+    this.vectorLayer.setStyle(
+      new Style({
+        stroke: new Stroke({
+          color: 'rgba(130,148,77, 0.8)',
+          width: 3,
+        }),
+        fill: new Fill({
+          color: 'rgba(130,148,77, 0.2)',
+        }),
+      }),
+    );
+  }
+
+  setGeoServerWMSUrl(geoServerUrl) {
+    this.tileLayer.getSource().setUrl(geoServerUrl);
+  }
+
+  getFeatureInfoUrl(coord) {
+    const projection = this.map.getView().getProjection();
+    const resolution = this.map.getView().getResolution();
+
+    if (this.tileLayer.getSource().getFeatureInfoUrl) {
+      return this.tileLayer
+        .getSource()
+        .getFeatureInfoUrl(coord, resolution, projection, {
+          info_format: 'application/json',
+          query_layers: this.tileLayer.getSource().getParams().layers,
+        });
+    }
+    return false;
+  }
+
+  getFeatureInfoAtCoordinate(coordinate) {
+    this.abortController.abort();
+    this.abortController = new AbortController();
+    const { signal } = this.abortController;
+    return fetch(this.getFeatureInfoUrl(coordinate), { signal })
+      .then((resp) => resp.json())
+      .then((r) => r.features)
+      .then((data) => {
+        const dataFeats = data.map((feat) => {
+          const multipolygon = [];
+          // Read the features, transform their coordinates to highlight the hovered feature.
+          feat.geometry.coordinates.forEach((coords) => {
+            const polygon = coords[0].map((c) => {
+              const transformedCoord = transform(
+                c,
+                this.dataProjection,
+                this.featureProjection,
+              );
+              return transformedCoord;
+            });
+            multipolygon.push([polygon]);
+          });
+          const newFeature = new Feature({
+            geometry: new MultiPolygon(multipolygon),
+          });
+          newFeature.setId(feat.id);
+          newFeature.setProperties(feat.properties);
+          return newFeature;
+        });
```

Comment:
You can do better than this. Probably in 1 or 2 lines of code.

Reply:
GeoJSON FeatureCollection to an  Array<ol.Feature>

Reply:
I initially tried with: https://openlayers.org/en/latest/apidoc/module-ol_format_GeoJSON-GeoJSON.html#readFeatures
But didn't made it work properly, I will try again :)

Reply:
Shorten it with:
`data.map((d) => this.format.readFeature(d));`